google/cloud/sql/connector/connector.py

```diff
@@ -284,6 +284,8 @@ def close(self) -> None:
         )
         # Will attempt to safely shut down tasks for 5s
         close_future.result(timeout=5)
+        # stop event loop (killing daemon thread)
+        self._loop.call_soon_threadsafe(self._loop.stop)
```

Comment:
1. I don't think we want to close the loop when it's been handed in by someone else (i.e. such as in the Instance constructor or in async land) We should probably only close if Thread=None. Also, we might want to exit the thread as well? 

2. I think because we are scheduling with call_soon_threadsafe that this hasn't actually happened by the time `close` has exited. Are we sure this is the behavior we want? Since we are calling.result above, I think that the implicit contract might be that when close() ends, the resources are cleaned up.

Reply:
How does this look? Looped over 100 `with Connector():` lines and only 1 thread was ever running at a time:
```python
# if background thread exists for Connector, clean it up
if self._thread:
    # stop event loop running in background thread
    self._loop.call_soon_threadsafe(self._loop.stop)
    # wait for thread to finish closing (i.e. loop to stop)
    self._thread.join()
```